Watchman.Common/Models/TimeRange.cs

```diff
@@ -36,27 +39,101 @@ public static TimeRange FromNow(DateTime end)
             return new TimeRange(DateTime.UtcNow, end);
         }
 
-        public void ForeachMinute(Action<int, DateTime> action)
+        public bool Contains(DateTime dateTime)
+        {
+            return dateTime >= this.Start && dateTime <= this.End;
+        }
+
+        public IEnumerable<TimeRange> MoveWhile(Func<TimeRange, bool> shouldContinue, Func<TimeRange, TimeSpan> getTime)
+        {
+            return this.MoveWhile(shouldContinue, getTime.Invoke(this));
+        }
+
+        public IEnumerable<TimeRange> MoveWhile(Func<TimeRange, bool> shouldContinue, TimeSpan time)
+        {
+            var timeRange = this;
+            while(shouldContinue.Invoke(timeRange))
+            {
+                yield return timeRange;
+                timeRange = timeRange.Move(time);
+            }
+        }
+
+        public TimeRange Move(TimeSpan time)
+        {
+            var clone = this.Clone();
+            clone.Start = this.Start.Add(time);
+            clone.End = this.End.Add(time);
+            return clone;
+        }
+
+        public TimeRange Clone()
+        {
+            return new TimeRange(this.Start, this.End);
+        }
+
+        public override string ToString()
+        {
+            var timezone = TimeZoneInfo.Local;
+            if (this.DaysBetween <= 7)
+            {
+                return $"{this.Start.ToLocalTime():dd/MM/yyyy HH:mm} - {this.End.ToLocalTime():dd/MM/yyyy HH:mm} (UTC+{timezone.BaseUtcOffset.TotalHours}:00)";
```

Comment:
dlaczego nie skorzystasz z ToLocalTimeString?

Reply:
no nie wiem czy łączenie `Watchman.Common` z frameworkiem to dobry pomysł